crates/oxc_linter/src/rules/eslint/no_empty.rs

```diff
@@ -49,33 +49,94 @@ impl Rule for NoEmpty {
+                ctx.diagnostic_with_suggestion(no_empty_diagnostic("block", block.span), |fixer| {
+                    if let Some(parent) = parent {
+                        if matches!(parent, AstKind::CatchClause(_)) {
+                            fixer.noop()
+                        } else {
+                            fixer.delete(&parent)
+                        }
+                    } else {
+                        fixer.noop()
+                    }
+                });
             }
             // The visitor does not visit the `BlockStatement` inside the `FinallyClause`.
             // See `Visit::visit_finally_clause`.
             AstKind::FinallyClause(finally_clause) if finally_clause.body.is_empty() => {
                 if ctx.semantic().trivias().has_comments_between(finally_clause.span) {
                     return;
                 }
-                ctx.diagnostic(no_empty_diagnostic("block", finally_clause.span));
+                ctx.diagnostic_with_suggestion(
+                    no_empty_diagnostic("block", finally_clause.span),
+                    |fixer| {
+                        let parent = ctx
+                            .nodes()
+                            .parent_kind(node.id())
+                            .expect("finally clauses must have a parent node");
+
+                        let AstKind::TryStatement(parent) = parent else {
+                            unreachable!("finally clauses must be children of a try statement");
+                        };
+
+                        // if there's no `catch`, we can't remove the `finally` block
+                        if parent.handler.is_none() {
+                            return fixer.noop();
+                        }
+
+                        if let Some(finally_kw_start) = find_finally_start(ctx, finally_clause) {
+                            fixer.delete_range(Span::new(finally_kw_start, finally_clause.span.end))
+                        } else {
+                            fixer.noop()
+                        }
+                    },
+                );
             }
             AstKind::SwitchStatement(switch) if switch.cases.is_empty() => {
-                ctx.diagnostic(no_empty_diagnostic("switch", switch.span));
+                ctx.diagnostic_with_suggestion(
+                    no_empty_diagnostic("switch", switch.span),
+                    |fixer| fixer.delete(switch),
+                );
             }
             _ => {}
         }
     }
 }
 
+fn find_finally_start(ctx: &LintContext, finally_clause: &BlockStatement) -> Option<u32> {
+    let src = ctx.source_text();
+    let finally_start = finally_clause.span.start as usize - 1;
+    let mut start = finally_start;
+
+    let src_chars: Vec<char> = src.chars().collect();
+
+    while start > 0 {
+        if let Some(&ch) = src_chars.get(start) {
+            if !ch.is_whitespace() {
+                if ch == 'y' && start >= 6 {
```

Comment:
Why 6? Will you make this a constant instead of a magic number?

Reply:
i don't think my braki was working properly when i wrote this.

the `6` was to make sure that on the below line, we don't try to do an out of bounds access.

regardles i rewrite this block - please take another look